core/trino-main/src/main/java/io/trino/operator/scalar/MathFunctions.java

```diff
@@ -914,7 +914,7 @@ public static long roundNShort(
                 @SqlType("decimal(p, s)") long num,
                 @SqlType(StandardTypes.INTEGER) long decimals)
         {
-            if (num == 0 || numPrecision - numScale + decimals <= 0) {
+            if (num == 0 || numPrecision - numScale + decimals < 0) {
```

Comment:
Can we have a test?

Reply:
UT added, https://github.com/trinodb/trino/pull/9371/files#diff-2dd37661d62217926a0672e367038d48e6fc38128009400b81107ec81c60a22bR955